stackn/stackn.py

```diff
@@ -817,6 +819,113 @@ def delete_meta_resource(
+        result = controller_id
+        params = {"app__category": "network", "app__slug": "reducer"}
+
+        if not result:
+            app_instances = call_project_endpoint(
+                "appinstances", params=params, conf={"STACKN_SECURE": secure}
+            )
+
+            if len(app_instances) == 0:
+                print("No controller apps found.")
+                return False
+
+            if len(app_instances) == 1:
+                result = app_instances[0]["id"]
+                return result
+
+            for app_instance in app_instances:
+                display_str = (
+                    f"Name: {app_instance['name']}, ID: {app_instance['id']}"
+                )
+                print(display_str)
+
+            while not result:
+                input_response = input("Enter the ID of the controller app: ")
+
+                for app in app_instances:
+                    if str(app["id"]) == input_response:
+                        result = input_response
+                        break
+
+        return result
+
+    def get_role():
+        result = role
+
+        if not result:
+            client_roles = call_project_endpoint(
+                "client_roles", conf={"STACKN_SECURE": secure}
+            )
+
+            client_roles_names = []
+
+            for client_role in client_roles:
+                name = client_role["name"]
+
+                client_roles_names.append(name)
+                display_str = f"Role: {name}"
+                print(display_str)
+
+            while not result:
+                input_response = input("Enter the role of the client: ")
+
+                if input_response in client_roles_names:
+                    result = input_response
+                else:
+                    role_names_str = " or ".join(client_roles_names)
+                    display_str = (
+                        f"Invalid role. Please enter {role_names_str}"
+                    )
+                    print(display_str)
+
+        return result
+
+    conf = {
+        "STACKN_PROJECT": project,
+        "STACKN_URL": studio_url,
+        "STACKN_SECURE": secure,
+    }
+
+    controller_id = get_reducer_id()
+
+    if not controller_id:
+        return False
+
+    print("Using controller app with ID: {}".format(controller_id))
+
+    role = get_role()
+
+    if not role:
+        return False
+
+    print("Using role: {}".format(role))
+
+    conf, auth_header, url = setup_project_endpoint_call(conf, "clients")
+
+    if not conf or not auth_header or not url:
+        print("Failed to setup project API endpoint.")
+        return False
+
+    data = {"name": name, "controller": controller_id, "role": role}
+
+    res = requests.post(
+        url, headers=auth_header, json=data, verify=conf["STACKN_SECURE"]
+    )
+
+    if res:
+        print(res.text)
+    else:
+        print("Failed to create client.")
+        print("Status code: {}".format(res.status_code))
+        print(res.text)
```

Comment:
return False

Reply:
Updated so that function never returns a value